pkg/operator/ceph/cluster/mgr/dashboard.go

```diff
@@ -110,52 +104,62 @@ func (c *Cluster) configureDashboardModules() error {
 	return nil
 }
 
-func (c *Cluster) configureDashboardModuleSettings(daemonID string) (bool, error) {
+func (c *Cluster) configureDashboardModuleSettings() (bool, error) {
+
 	monStore := config.GetMonStore(c.context, c.clusterInfo)
 
-	daemonID = fmt.Sprintf("mgr.%s", daemonID)
+	// Remove any existing per mgr-daemon configuration
+	for _, daemonID := range c.getDaemonIDs() {
```

Comment:
Before calling all these individual commands, how about if we do a sing `monStore.Get()` to figure out if one of them is set (perhaps the last one in the list), and then only call all these deletes if it is set. Then after the upgrade, future reconciles will not perform so many unnecessary deletes. It's just a small optimization, but one that adds up.

Reply:
@travisn It's indeed a good idea, unfortunately the get always work even when the entity doesn't exist. For example, let's get the field of the invalid "mgr.z":

```
bash-4.4$ ceph config get mgr.z "mgr/dashboard/server_port"
7777
```
ceph config for whatever reason returns the default values even in this case... sounds like BUG but it could be a "feature" to prevent something from crashing in case the config section doesn't exist.

Reply:
Oh right, we can't detect if the setting applies to a specific daemon. 

Then I'd suggest just setting a bool so that it's only done once per time the operator runs. It's fine if the operator restarts and runs the deletes again, but just trying to avoid the deletes with every reconcile.

Reply:
Done